### ISSUE TITLE
M3u8 live streaming error: PlaylistStuckException

### ISSUE THREAD

**Issue by Shreyansh1995** [google/ExoPlayer#8336]
I was trying to play Live Streaming m3u8 URL but not able to play for more than 40 seconds. It's throwing following error. Kindly look into this matter.

URL: https://aajtaklive-amd.akamaized.net//hls//live//2014416//aajtak//aajtaklive//aajtak_5//chunklist.m3u8

Platform Info:
Device: OnePlus GM1901 (OnePlus7)
Board: msmnile
Type&Tags: user (release-keys)
Android 10 SDK 29 (REL)
ABIs: arm64-v8a, armeabi-v7a, armeabi, 

Stacktrace:
```
com.google.android.exoplayer2.ExoPlaybackException: com.google.android.exoplayer2.source.hls.playlist.HlsPlaylistTracker$PlaylistStuckException
	at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:386)
	at android.os.Handler.dispatchMessage(Handler.java:103)
	at android.os.Looper.loop(Looper.java:214)
	at android.os.HandlerThread.run(HandlerThread.java:67)
Caused by: com.google.android.exoplayer2.source.hls.playlist.HlsPlaylistTracker$PlaylistStuckException
	at com.google.android.exoplayer2.source.hls.playlist.DefaultHlsPlaylistTracker$MediaPlaylistBundle.processLoadedPlaylist(DefaultHlsPlaylistTracker.java:637)
	at com.google.android.exoplayer2.source.hls.playlist.DefaultHlsPlaylistTracker$MediaPlaylistBundle.onLoadCompleted(DefaultHlsPlaylistTracker.java:518)
	at com.google.android.exoplayer2.source.hls.playlist.DefaultHlsPlaylistTracker$MediaPlaylistBundle.onLoadCompleted(DefaultHlsPlaylistTracker.java:443)
	at com.google.android.exoplayer2.upstream.Loader$LoadTask.handleMessage(Loader.java:450)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	... 2 more
```

**Comment by xiaoxiaoliu6** [comment]
https://aajtaklive-amd.akamaized.net//hls//live//2014416//aajtak//aajtaklive//aajtak_5//chunklist.m3u8   
There is something wrong with this m3u8 file

**Comment by stevemayhew** [comment]
@Shreyansh1995  this is likely an issue with the origin server.  ExoPlayer reports this exception when the live playlist does not update within the required time.  In greater detail:

The player considers the following variables for detecting the playlist has updated:

1. The *Media Sequence Number* (MSN), is used to keep track of what segments have been removed from the live playlist, the origin responsibility is defined in [Section 6.2.2 Live Playlists](https://tools.ietf.org/html/draft-pantos-hls-rfc8216bis-04#section-6.2.2) first paragraph
2. The segment count (SegCnt) (this will increase for type EVENT live playlists, for regular live it stays the same as window size)
3. If the live or EVENT playback is ended (when origin adds an <code>EXT-X-END</code> to a live playlist)

The player polls the playlist for updates (according to the rules in [Pantos Section 6.4.4](https://tools.ietf.org/html/draft-pantos-hls-rfc8216bis-04#section-6.3.4) and checks if the playlist updated.  

Each time the player re-loads the playlist it check for *playlist updated*, that is one of these three case must be true:

1. MSN<sub>Reloaded Playlist</sub> > MSN<sub>Previous Playlist</sub>
2. SegCnt<sub>Reloaded Playlist</sub> > SegCnt<sub>Previous Playlist</sub>
3. EXT-X-END<sub>Reloaded Playlist</sub> is present and EXT-X-END<sub>Previous Playlist</sub> was not 

Case 1 is the normal live case (the window slides, old segments are removed),   Case 2 only occurs for live "EVENT", in that case old segments don't age out.  Case 3 is when the live EVENT ends.   Note, after case 3 the player stops reloading the playlist (so trivially it cannot be "stuck" after this)

**Comment by Shreyansh1995** [comment]
Hi @stevemayhew  thanks for the reply. 
But the same link is working perfectly with mx player, x player and vlc player. 
How to fix this issue in my app with same link. Any suggestion?

**Comment by Shreyansh1995** [comment]
@xiaoxiaoliu6 but this url working fine in Mx player, VLC player and Xplayer. And i have many url which is working fine on other players but throwing exception in exoplayer

**Comment by xiaoxiaoliu6** [comment]
@Shreyansh1995 your url working not fine in vlc player for mac os

**Comment by stevemayhew** [comment]
@Shreyansh1995  The URL returns a 403 for me, so I can't really help you further, but check the updated conditions from my previous comment, if they are not met then ExoPlayer will throw the `PlaylistStuckException`.   The other players you mention may simply freeze in buffering state rather than throwing an exception.   

The time is pretty generous,  assuming the playlist has `EXT-X-TARGETDURATION` of 6 seconds ExoPlayer would wait 18 seconds for new content from the origin before throwing the error.

**Comment by ojw28** [comment]
The responses above are correct. `PlaylistStuckException` indicates an issue where the media playlist isn't being updated correctly on the server-side. As @stevemayhew points out, the time ExoPlayer allows for an update to occur is pretty generous.

I was able to play the stream without any issues for a couple of minutes, and was not able to reproduce the failure. Given the mixed outcomes above, it could be that the issue with this stream is intermittent. Closing as `bad media`, since the issue here is most likely with the media itself.
